client/build.go

```diff
@@ -45,11 +45,16 @@ func (c *Client) Build(ctx context.Context, opt SolveOpt, product string, buildF
 	}
 
 	cb := func(ref string, s *session.Session) error {
-		g, err := grpcclient.New(ctx, feOpts, s.ID(), product, c.gatewayClientForBuild(ref), gworkers)
+		gwClient := c.gatewayClientForBuild(ref)
+		g, err := grpcclient.New(ctx, feOpts, s.ID(), product, gwClient, gworkers)
 		if err != nil {
 			return err
 		}
 
+		if c, ok := g.(gateway.Client); ok {
```

Comment:
I don't really like this cast in here. Looks easy to break if thing change.

I think instead wee should either define GRPCClient

```
type GrpcClient interface {
	client.Client
	Run(context.Context, client.BuildFunc) error
}
```

or wrap `BuildFunc` and set the caps inside the wrapper.

Reply:
Updated https://github.com/moby/buildkit/pull/1627/commits/dc3d45de8731e4dc4f9e7eabba4ed5f1799a3eaf

We could also just extend the `GrpcClient` with `BuildOpts() BuildOpts` if want to make the api usage explicit.